src/datasets/mod.rs

```diff
@@ -0,0 +1,195 @@
+use std::fmt::Debug;
+
+use rulinalg::matrix::Matrix;
+use rulinalg::vector::Vector;
+
+/// Dataset container
+#[derive(Clone, Debug)]
+pub struct Dataset<D, T> where D: Clone + Debug, T: Clone + Debug {
```

Comment:
I think this makes sense for now. We might want to be more strict in future if we want to be generic over `DataSets`. However, this is something that I don't think we will ever want to do.